examples/hello-world/job_api/pt/client_api_kmeans.py

```diff
@@ -0,0 +1,156 @@
+        "jet_2_eta",
+        "jet_2_phi",
+        "jet_2_b_tag",
+        "jet_3_pt",
+        "jet_3_eta",
+        "jet_3_phi",
+        "jet_3_b_tag",
+        "jet_4_pt",
+        "jet_4_eta",
+        "jet_4_phi",
+        "jet_4_b_tag",
+        "m_jj",
+        "m_jjj",
+        "m_lv",
+        "m_jlv",
+        "m_bb",
+        "m_wbb",
+        "m_wwbb",
+    ]
+
+    # Specify the file path
+    file_path = os.path.join(data_input_dir, "headers.csv")
+
+    with open(file_path, "w", newline="") as file:
+        csv_writer = csv.writer(file)
+        csv_writer.writerow(features)
+
+    print(f"features written to {file_path}")
+
+    # Split the data
+    split_higgs(
+        input_data_path=os.path.join(data_input_dir, "higgs.csv"),
+        input_header_path=os.path.join(data_input_dir, "headers.csv"),
+        output_dir=data_output_dir,
+        site_num=n_clients,
+        sample_rate=0.3,
+    )
+
+    # Create the federated learning job
+    job = FedJob(name="kmeans")
+
+    controller = ScatterAndGather(
+        min_clients=n_clients,
+        num_rounds=num_rounds,
+        wait_time_after_min_received=0,
+        aggregator_id="aggregator",
+        persistor_id="persistor",  # TODO: Allow adding python objects rather than ids
+        shareable_generator_id="shareable_generator",
+        train_task_name="train",  # Client will start training once received such task.
+        train_timeout=0,
+    )
+    job.to(controller, "server")
+
+    # For kmeans with sklean, we need a custom persistor
+    # JoblibModelParamPersistor is a persistor which save/read the model to/from file with JobLib format.
+    persistor = JoblibModelParamPersistor(initial_params={"n_clusters": 2})
+    # When assigning the persistor to the server, we need to specify the id that's expected
+    # by the ScatterAndGather controller.
+    job.to(persistor, "server")
+
+    # Similarly, ScatterAndGather expects a "shareable_generator" which we need to assign to the server.
+    job.to(FullModelShareableGenerator(), "server")
+
+    # ScatterAndGather also expects an "aggregator" which we define here.
+    # The actual aggregation function is defined by an "assembler" to specify how to handle the collected updates.
+    aggregator = CollectAndAssembleAggregator(
+        assembler_id="kmeans_assembler"
+    )  # TODO: Allow adding KMeansAssembler() directly
+    job.to(aggregator, "server")
+
+    # This is the assembler designed for k-Means algorithm.
+    # As CollectAndAssembleAggregator expects an assembler_id, we need to specify it here.
+    job.to(KMeansAssembler(), "server", id="kmeans_assembler")
+
+    # Add clients
+    for i in range(n_clients):
+        executor = ScriptExecutor(
+            task_script_path=train_script,
+            task_script_args=f"--data_root_dir {data_output_dir}",
+            params_exchange_format=ExchangeFormat.RAW,  # kmeans requires raw values only rather than PyTorch Tensors (the default)
```

Comment:
this is raw so we should move it outside of "job_api/pt"

Reply:
Moved to `job_api/sklearn`